src/solver/infeasible-problem-analysis/problem.h

```diff
@@ -11,22 +11,77 @@ namespace Antares
 {
 namespace Optimization
 {
-class InfeasibleProblemAnalysis
+
+
+class SingleAnalysis
 {
 public:
-    InfeasibleProblemAnalysis() = delete;
-    explicit InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx);
-    InfeasibleProblemReport produceReport();
+    SingleAnalysis(std::shared_ptr<operations_research::MPSolver> problem_);
```

Comment:
no need for shared ownership here, reference or raw pointer seems more appropriate

Reply:
Done